dev/App.js

```diff
@@ -59,15 +65,32 @@ class App extends Component {
     )
       .then(response => response.json())
       .then(figure => {
-        this.setState({
-          currentMockIndex: mockIndex,
-          data: figure.data,
-          layout: figure.layout,
-          frames: figure.frames,
-        });
+        const {data, layout, frames} = figure;
+        this.updateState(data, layout, frames, mockIndex);
       });
   }
 
+  updateState(data, layout, frames, currentMockIndex) {
+    this.setState({
+      data,
+      layout,
+      frames,
+      currentMockIndex,
+      full: 'hit refresh',
```

Comment:
there's `mockIndex` in the state above, are `mockIndex` and `currentMockIndex` meant to be the same?

Reply:
The state key is indeed `currentMockIndex`, so this is correct, but in a function above we get a new index called `mockIndex`, which we then use to overwrite.